tests/api_tests/base/test_serializers.py

```diff
@@ -1,13 +1,92 @@
 # -*- coding: utf-8 -*-
+from api.base.wsgi import application
+
 from nose.tools import *  # flake8: noqa
 import httplib as http
+import contextlib
+import mock
+
+from rest_framework import serializers as ser
 
 from tests.base import ApiTestCase
 from tests import factories
 
 from api.base.settings.defaults import API_BASE
+from api.base import serializers as base_serializers
 from api.nodes.serializers import NodeSerializer
 
+FakeModel = mock.Mock
+
+class FakeSerializer(base_serializers.JSONAPISerializer):
+
+    class Meta:
+        type_ = 'foos'
+
+    links = base_serializers.LinksField({
+        'null_field': 'null_field',
+        'valued_field': 'valued_field',
+        'null_wb_field': base_serializers.WaterbutlerLink(kind='file', null=True),
+        'valued_wb_field':  base_serializers.WaterbutlerLink(kind='file')
+    })
+    
+    null_link_field = base_serializers.JSONAPIHyperlinkedIdentityField(
+        'null:bar',
+        lookup_field='foo',
+        lookup_url_kwarg='bar',
+        link_type='related'
+    )
+    valued_link_field = base_serializers.JSONAPIHyperlinkedIdentityField(
+        'foo:bar',
+        lookup_field='foo',
+        lookup_url_kwarg='bar',
+        link_type='related'
+    )
+
+    def null_field(*args, **kwargs):
+        return None
+
+    def valued_field(*args, **kwargs):
+        return 'http://foo.com'
+
+def mock_hyperlink(instance, view_name, *args, **kwargs):
+    if view_name == 'null:bar':
+        return None
+    else:
+        return 'http://foo.com'
+
+def mock_wb_link(*args, **kwargs):
+    return 'http://api.foo.com' if not kwargs.get('null', False) else None
+
+class TestNullLinks(ApiTestCase):
+
+    def test_null_links_are_omitted(self):
+        ser = FakeSerializer()
+        with contextlib.nested(
```

Comment:
With statements support multiple context managers; no need to use `nested` in 2.7.

https://docs.python.org/2/reference/compound_stmts.html#the-with-statement

```python
with A() as a, B() as b:
    suite
```